test/extended/apiserver/latency.go

```diff
@@ -14,38 +14,41 @@ import (
 	promhelper "github.com/openshift/origin/test/extended/util/prometheus"
 )
 
+const (
+	apiserverReadVerbs  = "GET|LIST"
+	apiserverWriteVerbs = "POST|PUT|PATCH|DELETE"
+)
+
 var _ = g.Describe("[sig-instrumentation][Late] API Server latency", func() {
 	defer g.GinkgoRecover()
 
 	oc := exutil.NewCLIWithoutNamespace("apiserver-latency")
 
-	// TODO: refine latency thresholds based on actual CI run results
-	//
+	// Latency thresholds defined below are based on the following CI jobs:
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26739/pull-ci-openshift-origin-master-e2e-gcp/1481905976666755072
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26739/pull-ci-openshift-origin-master-e2e-gcp/1481879797410828288
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26739/pull-ci-openshift-origin-master-e2e-aws-fips/1481879797377273856
+	// - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26739/pull-ci-openshift-origin-master-e2e-aws-fips/1481905976616423424
 	// Latency thresholds should be calculed based on the following queries:
-	// - histogram_quantile(0.99,sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",scope=~"resource|"}[$testDuration])) by (le,verb))
-	// - histogram_quantile(0.99,sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",scope="namespace"}[$testDuration])) by (le,verb))
-	// - histogram_quantile(0.99,sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",scope="cluster"}[$testDuration])) by (le,verb))
+	// - histogram_quantile(0.99,sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"GET|LIST"}[$testDuration])) by (le,scope))
+	// - histogram_quantile(0.99,sum(rate(apiserver_request_duration_seconds_bucket{job="apiserver",verb=~"POST|PUT|PATCH|DELETE"}[$testDuration])) by (le,scope))
 	g.It("should verify that the apiserver requests latency are within expected thresholds", func() {
 		promClient := oc.NewPrometheusClient(context.TODO())
 
 		// 99th percentile of resource-scoped requests latency
 		scope := "resource|"
-		expectRequestLatency(promClient, "GET", scope, parseDuration("100ms"))
-		expectRequestLatency(promClient, "POST", scope, parseDuration("100ms"))
-		expectRequestLatency(promClient, "PUT", scope, parseDuration("100ms"))
-		expectRequestLatency(promClient, "PATCH", scope, parseDuration("100ms"))
-		expectRequestLatency(promClient, "DELETE", scope, parseDuration("100ms"))
+		expectRequestLatency(promClient, apiserverReadVerbs, scope, parseDuration("200ms")) // got 130ms in CI
+		expectRequestLatency(promClient, apiserverWriteVerbs, scope, parseDuration("1s"))   // threshold defined by sig-scalability
```

Comment:
the point of this is to latch *our* thresholds based on our observed CI behavior.  Not to mirror something generic.

I'm willing to start nearly anywhere though.

Reply:
Updated